### PR TITLE
pushgateway: adds configuration for omitting job name

### PR DESCRIPTION
**Pull request by geofholbrook**
intended for compatibility with gravel gateway (https://github.com/sinkingpoint/prometheus-gravel-gateway)
see issue https://github.com/siimon/prom-client/issues/502

### CODE REVIEW

File: test/pushgatewayTest.js [hunk 1]
```diff
@@ -96,6 +96,19 @@ describe.each([
 						expect(err.message).toStrictEqual('Pushgateway request timed out');
 					});
 			});
+
+			it('should be possible to configure for gravel gateway integration (no job name required in path)', async () => {
+				const mockHttp = nock('http://192.168.99.100:9091')
+					.post('/metrics', body)
+					.reply(200);
+
+				instance.requireJobName = false;
```
Comment (SimenB) [inline test/pushgatewayTest.js:105]: instead of mutating `instance`, create a new one?

Comment (geofholbrook) [reply test/pushgatewayTest.js]: Wanted to do that, but for some reason I failed to get passing tests on my first try. Just committed my second try.

**Review by zbjornson — APPROVED** [review]
Great, thanks. I added a changelog entry and rebased in my last push.

**Comment by SimenB** [comment]
https://github.com/siimon/prom-client/releases/tag/v15.1.0

**Pull request merged**